src/lib/expansion/accordion.ts

```diff
@@ -0,0 +1,120 @@
+import {Subscription} from 'rxjs/Subscription';
+import {Subject} from 'rxjs/Subject';
+import 'rxjs/add/operator/takeUntil';
+
+/** Used to generate unique ID for each expansion panel. */
+let nextId = 0;
+
+export class CdkAccordionChild implements OnDestroy {
+  /** Event emitted every time the MdAccordianChild is closed. */
+  @Output() close = new EventEmitter<null>();
+  /** Event emitted every time the MdAccordianChild is opened. */
+  @Output() open = new EventEmitter<null>();
+  /** Event emitted when the MdAccordianChild is destroyed. */
+  @Output() destroy = new EventEmitter<null>();
+  /** Whether the MdAccordianChild is expanded. */
+  expanded: boolean;
+
+  /** The unique MdAccordianChild id. */
+  private _id: number = nextId++;
+  get id(): string { return `cdk-accordion-child-${this._id}`; }
+  /** Emits an event for the accordion child being destroyed. */
+  ngOnDestroy() {
+    this.destroy.emit();
+  }
+}
+
+@Directive({
+  selector: '[cdk-accordion]',
+})
+export class CdkAccordion implements AfterContentInit, OnDestroy {
+  /** Whether the expansion indicator should be shown. */
+  private  _showIndicator: boolean = true;
+  @Input()
+  get showExpandIndicator(): boolean { return this._showIndicator; }
+  set showExpandIndicator(show: boolean) { this._showIndicator = coerceBooleanProperty(show); }
+  /** Whether the panel set should use flat styling. */
+  @Input() accordionStyle: string = 'default';
+  /** Whether the panel set should allow multiple open panels. */
+  private  _multi: boolean = false;
+  @Input()
+  get multi(): boolean { return this._multi; }
+  set multi(multi: boolean) { this._multi = coerceBooleanProperty(multi); }
+
+  /** A subject to be completed on destroy to clean up subscriptions. */
+  private _destroySubject = new Subject();
+  /** A list of subscriptions for panel open events. */
+  private _childrenSet: Set<CdkAccordionChild> = new Set();
+  /** QueryList of all expansion panels in the expansion panel set. */
+  @ContentChildren(CdkAccordionChild) private _children: QueryList<CdkAccordionChild>;
+
+  /** Set up event subscriptions for all panels for when they open, closing the other panels. */
+  ngAfterContentInit() {
+    this.registerChildren(this._children);
+    this._children.changes
+      .takeUntil(this._destroySubject)
+      .subscribe((children: QueryList<CdkAccordionChild>) => {
+        this.registerChildren(children);
+      });
+  }
+
+  /** Clean up panel subscriptions. */
+  ngOnDestroy() {
+    this._destroySubject.next();
+    this._destroySubject.complete();
+  }
+
+  /** Registers a QueryList of panels. */
+  private registerChildren(panels: QueryList<CdkAccordionChild>) {
+    this._children.forEach(panel => this.registerChild(panel));
+  }
+
+  /**
+   * Registers a CdkAccordionChild panel if has not already been registered.  Creates subscriptions
+   * for the open and destroy events for the panel.
+   */
+  private registerChild(child: CdkAccordionChild) {
```

Comment:
Rather than going through all of this work to have the accordion manage which panels are open and closed, why not have the individual items do this using `UniqueSelectionDispatcher`? I think that would be significantly less code.